config/config.py

```diff
@@ -0,0 +1,187 @@
+
+	@property
+	def simulator_config(self):
+		return self.__remote_simulator_config
+
+
+class _RemoteSimulatorConfigDAO:
+	"""
+	Simulator Config DAO
+	"""
+	KEY_ENABLED = "enabled"
+	KEY_PORT = "port"
+
+	def __init__(self, simulator):
+		self.__enabled = simulator[self.KEY_ENABLED]
+		self.__port = simulator[self.KEY_PORT]
+
+	@property
+	def enabled(self):
+		return self.__enabled
+
+	@property
+	def port(self):
+		return self.__port
+
+
+class _ConfigDAO:
+	"""
+	Data Access Object (DAO) for the configuration
+	"""
+	KEY_DEBUG_CONFIG = "debug"
+	KEY_DESK_CONFIGS = "desks"
+
+	def __init__(self, loader):
+		self.__loader = loader
+		self.__properties = loader.properties
+
+		self.__desks = []
+		self.__debug_config = _DebugConfigDAO(self.__properties[self.KEY_DEBUG_CONFIG])
+
+		for desk in self.__properties[self.KEY_DESK_CONFIGS]:
+			self.__desks.append(_DeskConfigDAO(desk))
+
+	@property
+	def debug_config(self):
+		return self.__debug_config
+
+	@property
+	def desks(self):
+		return self.__desks
+
+	def __str__(self):
+		return self.__properties.__str__()
+
+	def update_desk_list(self, new_desk):
+		updated_desks = self.__loader.update_desk_list(new_desk)
+		self.__desks = list(map(lambda desk_dict: _DeskConfigDAO(desk_dict), updated_desks))
+
+		return updated_desks
+
+class __YAMLConfigLoader:
+	"""
+	YAML Configuration Loader
+	"""
+
+	def __init__(self, yaml_file):
+		"""
+		Loads YAML config file
+		:param yaml_file: Path to YAML file
+		"""
+
+		self.__property_dictionary = {}
+		self.__yaml_file = yaml_file
+		self.load_config()
+
+	def load_config(self):
+		"""
+		Loads YAML config file.
+		:return: None
+		"""
+
+		# Load the file as stream
+		with open(self.__yaml_file, 'r') as stream:
+			try:
+				# Parse YAML. The yaml.load function already puts it into a dictionary, so no conversion is necessary
+				self.__property_dictionary = yaml.load(stream)
+			except yaml.YAMLError as e:
+				print(e)
+				raise e
+
+	def update_desk_list(self, desk_dict):
+		self.load_config()
+		self.properties['desks'].append(desk_dict)
+
+		self.__write_to_config(self.properties)
+
+		return self.properties['desks']
+
+	def __write_to_config(self, new_config):
+		with open(self.__yaml_file, 'w') as stream:
+			try:
+				stream.write(yaml.dump(new_config))
+			except yaml.YAMLError as e:
+				print(e)
+				raise e
+
+	@property
+	def properties(self):
+		"""
+		Get property dictionary
+		:return: Property Dictionary
+		"""
+
+		return self.__property_dictionary
+
+
+# Global Config DAO available for access via import
+# Should be initialized when this module is imported
+CONFIG: _ConfigDAO = None
+
+def load_config(file_name="config/config.yaml"):
+	"""
+	This method ensures that we only load the config once.
+	:return: The global Config DAO
+	"""
+	global CONFIG
+
+	if not CONFIG:
+
+		CONFIG = _ConfigDAO(__YAMLConfigLoader(file_name))
+
+		if CONFIG.debug_config.enabled:
+			#Print config
+			print("[Config]:", CONFIG)
+
+	return CONFIG
```

Comment:
Could do `return CONFIG or _ConfigDAO(__YAMLConfigLoader(file_name))` if:
```
if CONFIG.debug_config.enabled:
    #Print config
    print("[Config]:", CONFIG)
```
was moved to __YAMLConfigLoader. Maybe it's better for the ConfigLoader to be responsible for printing the config anyways.

Reply:
I'll make these changes in a different PR since because this one was branched off of https://github.com/Smart-Library/desk-controller/pull/3, it was showing changes for some files that I didn't touch